packages/Link/src/Link.styles.ts

```diff
@@ -0,0 +1,102 @@
+import stylers from "@paprika/stylers";
+import styled, { css } from "styled-components";
+import tokens from "@paprika/tokens";
+import { NewTab as NewTabIcon } from "@paprika/icon";
+
+const noUnderlineStyles = css`
+  text-decoration: none;
+`;
+
+const blackFont = css`
+  color: ${tokens.textColor.default};
+  &:hover,
+  &:focus {
+    color: ${tokens.textColor.linkHover};
+  }
+`;
+
+const whiteFont = css`
+  color: ${tokens.color.offWhite};
+  ${noUnderlineStyles};
+  &:hover,
+  &:focus {
+    color: ${tokens.color.offWhite};
+  }
+`;
+
+const linkHoverStyles = css`
+  &:hover,
+  &:focus {
+    color: ${tokens.textColor.linkHover};
+    text-decoration: underline;
+  }
+`;
+
+/** Navigation */
+const navigationStyles = css`
+  display: flex;
+  ${blackFont};
+  padding: 0 calc(3 * ${tokens.spaceSm});
+
+  &,
+  &:hover,
+  &:focus {
+    text-decoration: none;
+    color: ${tokens.textColor.default};
+  }
+  &:hover {
+    background-color: ${tokens.color.blackLighten70};
+  }
+`;
+
+export const LinkContent = styled.span`
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+`;
+
+export const ExternalLinkIconStyles = styled(NewTabIcon)`
+  color: ${tokens.textColor.icon};
+  display: inline-block;
+  font-size: 20px;
+  &[data-pka-anchor="icon"] {
+    margin-right: 0;
+  }
+`;
+
+export const Link = styled.a<{
+  isSubtle: boolean;
+  hasNoUnderline: boolean;
+  isDark: boolean;
+  isMenu: boolean;
+  maxWidth: string;
+  [x: string]: any;
+}>(
+  ({ hasNoUnderline, isSubtle, isDark, isMenu, maxWidth }) => css`
+    &[data-pka-anchor="link"] {
+      align-items: center;
+      border-radius: ${tokens.border.radius};
+      color: ${tokens.textColor.link};
+      display: inline-flex;
+      left: -${tokens.spaceSm};
+      ${maxWidth ? `max-width: ${maxWidth};` : ""}
+      padding: 1px ${tokens.spaceSm};
+      position: relative;
+
+      &:focus {
+        outline: none;
+      }
+
+      html:not([data-whatinput="mouse"]) &:focus,
+      &[data-has-forced-focus]:focus {
+        ${stylers.focusRing()}
```

Comment:
Maybe we should apply a `stylers.zIndex(1)` here as well, to avoid this kind of situation:
![Screen Shot 2021-10-21 at 12 12 21 PM](https://user-images.githubusercontent.com/14944896/138341965-d3965d46-1227-4775-a164-9487c0be4b7f.png)

Reply:
Should we use the inset shadow focus ring if we really want to avoid this? Sometimes I'm confused when I should use the inset shadow ring or outside shadow ring. In the screenshot, there's a lot of padding space, so maybe inset make sense?

Reply:
The padding space does look really large 🤔
`stylers.z(1)`:
<img width="300" alt="Screen Shot 2021-10-21 at 5 22 23 PM" src="https://user-images.githubusercontent.com/92059030/138374357-4980a808-bc03-41ec-b6f5-cbc78b3a798b.png">
`stylers.focusRing(true)`:
<img width="300" alt="Screen Shot 2021-10-21 at 5 21 17 PM" src="https://user-images.githubusercontent.com/92059030/138374354-ccc7926a-45fa-401c-9ae2-839ae5f1766b.png">

In this example it's using margin to avoid this, but I don't think we should set a default margin...?
<img width="350" alt="Screen Shot 2021-10-21 at 5 26 37 PM" src="https://user-images.githubusercontent.com/92059030/138374458-f3ba3f9d-58ed-4d31-8e23-d018be6e2e82.png">

Reply:
I agree with Mike and if the state effects are full width then it makes sense to have squared corners. Rounded corners are better if there is a padding on the left and right (as in the screenshots above)